docs/alincome/expenses.md

```diff
@@ -0,0 +1,271 @@
+
+  ![A screenshot of a webpage. At the top, it asks "What kind of expenses do you have?" There is a checkbox next to each option, like "rent" and "food". There is a continue button at the bottom](../assets/alincome_expenses_checkboxes.jpg)
+
+  :::tip `display_name`
+
+  As mentioned in the previous section, `expense_terms` provides the `ALExpense` objects in your `ALExpenseList` with the values for `source` and for `display_name`; everything to the left of the colon on each line is `source` and everything to the right is the longer `display_name`.
+
+  :::
+
+* For each expense that the user said they have, there will be one question screen,
+  asking for how often they pay for that expense, and how much they pay. If they selected "other", they will also
+  be prompted to enter the name of that expense.
+
+  If you want to change the options for how often they pay the amount, you can define a `times_per_year_list`,
+  similarly to the `expense_terms` above. By default, the list contains "weekly", "once every two weeks", "twice per month",
+  "monthly", "once every 3 months", "once every 6 months", and "yearly". For this tutorial, let's make that list a little shorter.
+  Add the below block to your code:
+
+  ```yml
+  variable name: times_per_year_list
+  data:
+    - [52, "Weekly"]
+    - [26, "Once every two weeks"]
+    - [24, "Twice per month"]
+    - [12, "Monthly"]
+    - [1, "Yearly"]
+  ```
+
+You can show let the user review the expenses they've entered
+with the following question block:
+
+```yml
+id: expenses review
+question: |
+  Review your expenses
+subquestion: |
+  Here is what you've told us about your monthly expenses. You can add more or continue
+  to the next question.
+
+  ${ expenses.table }
+
+  ${ expenses.add_action() }
+field: review_expenses_screen
+```
+
+:::tip Review screens
+You can also add the expense list to a [review screen](https://docassemble.org/docs/fields.html#review) with the below code:
+
+```yml
+review:
+  - Edit: expenses.revisit
+    button: |
+      **Expenses**:
+
+      % for item in expenses:
+      * ${ item.display_name }: ${ currency(item.value) }
+      % endfor
+```
+
+:::
+
+That's all you need to gather the information, but the ALExpenseList class also
+gives you many ways to display the info. This is the same way of displaying the `ALIncomeList` class, so if you've used `ALIncomeList` before feel free to skim through this section.
+
+Let's add one last screen to display our information. We'll
+describe what each line of code is doing on the screen.
+
+```yml
+event: final_screen
+question: Summary of your expenses
+subquestion: |
+  All of the types of expenses, which we call "sources": ${ expenses.sources() }
+
+  Sum of all of the expenses (by default, this is annually): ${ currency(expenses.total()) }
+
+  Sum of all of the expenses over a month: ${ currency(expenses.total(times_per_year=12)) }
+
+  Value of just one expense: ${ currency(expenses.total(source="rent")) }
```

Comment:
I thought I remembered that multiple expenses could have the same source? If so, "Value of just one category of expense" and the description for "source" might need to be changed too. If not, never mind!

Reply:
Yes and no. Yes, `ALExpenseList` can have more than once expense from the same source. However the UX that we've landed on as the default (and is being used in this tutorial) won't have things from the same source, as it's showing a dict of checkboxes before hand, with each key for the checkbox as one source.

A user can select other and then add another expense from the same source, but TBH, that was more for resiliency and couldn't remember if there was a specific feature that someone would want to use with that.

Trying to balance being accurate and not confusing users with technical details they shouldn't have to know about to use the library is hard.  The description of source is accurate, and I don't think I would change it, even if we talk about allowing duplicates. I might change this line to be a bit more accurate though.

```suggestion
  Value of expenses with the source "rent": ${ currency(expenses.total(source="rent")) }
```